### PR TITLE
Add status overview in sidebar.

### PR DESCRIPTION
**Pull request by gijskant**
- Status overview for requests and lab requests;
- Interceptor for error handling;
- Updated some login/logout functionality.

### CODE REVIEW

File: src/main/resources/static/app/components/sidebar/lab-requests-sidebar-template.html [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        <li  ng-class="{active: activeSidebar === 'unclaimed'}">
+            <a ng-href="/#/lab-requests/unclaimed">{{'Unclaimed'|translate}} <span class="badge pull-right">{{unclaimedReqs.length}}</span></a>
+        </li>
+        <hr />
+        <li ng-repeat="status in statusesForRole"
+            ng-class="{active: activeSidebar === status}">
+            <a ng-href="/#/lab-requests/{{status}}">{{status|translate}} <span class="badge pull-right">{{requestsByStatus[status].length}}</span></a>
+        </li>
+    </ul>
+</div>
```
Comment (rnugraha) [inline src/main/resources/static/app/components/sidebar/lab-requests-sidebar-template.html]: can we maintain max 120 char width

File: src/main/resources/static/app/interceptors/ResponseInterceptor.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                case 302:
+                case 403:
+                    if (url === 'user') {
+                        console.log('Error fetching user data.');
+                        $rootScope.logErrorResponse(response);
+                        return $q.reject(response);
+                    }
+                    $rootScope.heartbeat().then(function(data) {
+                        // User is logged in, but denied access; redirect to start page.
+                        console.log('User is logged in, but denied access; redirect to start page.');
```
Comment (rnugraha) [inline src/main/resources/static/app/interceptors/ResponseInterceptor.js]: remove console

File: src/main/resources/static/app/interceptors/ResponseInterceptor.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                            placement : 'top-right',
+                            type : 'warning',
+                            show : true,
+                            duration : 15
+                        });
+                        $location.path('/');
+                        return $q.reject(response);
+                    }, function(error) {
+                        // User not logged in or session is expired; redirect to login page.
+                        console.log('User not logged in or session is expired; redirect to login page.');
```
Comment (rnugraha) [inline src/main/resources/static/app/interceptors/ResponseInterceptor.js]: remove console

File: src/main/resources/static/app/login/LoginController.js [hunk 4]
```diff
@@ -89,13 +72,16 @@ angular.module('ProcessApp.controllers')
...
-                                        redirectUrl = '/';
-                                    } else {
+                                    console.log("lab_user: " + $rootScope.isLabUser());
+                                    if ($rootScope.isLabUser() || $rootScope.isHubUser()) {
                                         redirectUrl = '/lab-requests';
+                                    } else {
+                                        redirectUrl = '/';
                                     }
                                 }
+                                console.log('Redirect to: ' + redirectUrl);
```
Comment (rnugraha) [inline src/main/resources/static/app/login/LoginController.js]: remove console

File: src/main/resources/static/app/services/FilterService.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright (C) 2016  Stichting PALGA
+ * This file is distributed under the GNU Affero General Public License
+ * (see accompanying file LICENSE).
+ */
+(function(console, _, angular) {
```
Comment (rnugraha) [inline src/main/resources/static/app/services/FilterService.js]: what is this console?

Comment (gijskant) [inline src/main/resources/static/app/services/FilterService.js]: Your browser console.

**Comment by gijskant** [comment]
@rnugraha Pull request updated. Merge?

**Pull request merged**